spec/unit/http/authenticator_spec.rb

```diff
@@ -19,13 +19,97 @@
 require "spec_helper"
 require "chef/http/authenticator"
 
+describe Chef::HTTP::Authenticator, :windows_only do
+  let(:class_instance) { Chef::HTTP::Authenticator.new(client_name: "test") }
+  let(:method) { "GET" }
+  let(:url) { URI("https://chef.example.com/organizations/test") }
+  let(:headers) { {} }
+  let(:data) { "" }
+  let(:node_name) { "test" }
+  let(:passwrd) { "some_insecure_password" }
+
+  before(:each) do
+    ::Chef::Config[:node_name] = "test"
+    cert_name = "chef-#{node_name}"
+    d = Time.now
+    end_date = Time.new(d.year, d.month + 3, d.day, d.hour, d.min, d.sec).utc.iso8601
+
+    my_client = Chef::Client.new
+    my_client.generate_pfx_package(cert_name, end_date)
+  end
+
+  after(:each) do
+    require "chef/mixin/powershell_exec"
+    extend Chef::Mixin::PowershellExec
+    cert_name = "chef-#{node_name}"
+    delete_certificate(cert_name)
+  end
+
+  context "when retrieving a certificate from the certificate store" do
+    it "retrieves a certificate password from the registry when the hive does not already exist" do
+      delete_registry_hive
+      expect { class_instance.get_cert_password }.not_to raise_error
+    end
+
+    it "should return a password of at least 14 characters in length" do
+      password = class_instance.get_cert_password
+      expect(password.length).to eql(14)
+    end
+
+    it "retrieves a certificate password from the registry when the hive exists" do
+      set_registry_hive
+      expect { class_instance.get_cert_password }.not_to raise_error
```

Comment:
Seems like `set_registry_hive` and `class_instance.get_cert_password` are basically the same thing?